zipline/finance/blotter.py

```diff
@@ -246,34 +246,37 @@ def get_transactions(self, bar_data):
             example, Zipline models the commission cost into the fill price
             of the transaction), then this is None.
         """
+
         closed_orders = []
         transactions = []
 
-        assets = self.asset_finder.retrieve_all(self.open_orders.keys())
-        asset_dict = {asset.sid: asset for asset in assets}
+        if len(self.open_orders) > 0:
+            assets = self.asset_finder.retrieve_all(self.open_orders.keys())
```

Comment:
`iter` over a dict will yield the keys, so I don't think we need to make a new list of them.  Or are we mutating them here?